cmd/agent/common/common_windows.go

```diff
@@ -138,6 +138,68 @@ func CheckAndUpgradeConfig() error {
 	return ImportConfig(DefaultConfPath, DefaultConfPath, false)
 }
 
+// SetInstallInfo imports installation information from Windows registry into
+// the install_info file. It leaves the file untouched if it already exists
+func SetInstallInfo() error {
+
+	// get install_info path
+	dataDir, err := winutil.GetProgramDataDir()
+	if err != nil {
+		return err
+	}
+	installInfoPath := filepath.Join(dataDir, "install_info")
+
+	// check that the file does not already exist
+	_, err = os.Stat(installInfoPath)
+	if err != nil && !os.IsNotExist(err) {
+		return fmt.Errorf("unable to stat %s: %s", installInfoPath, err)
+	} else if err == nil { // File already exists
+		return nil
+	}
+
+	// Get msiexec version from registry
+	k, err := registry.OpenKey(registry.LOCAL_MACHINE,
```

Comment:
it occurs to me that the install method is _always_ msi.  The question is, whether it's msi from the command line, msi launched from ansible, from chocolatey, etc.

Point is IMO we should _merge_ the existing information, so that we get the msiexec and any other msi information rather than just skip if the file was already created by ansible/chef/chocolately/etc